src/query/sortResults.ts

```diff
@@ -0,0 +1,91 @@
+import debug from 'debug'
+import get from 'lodash.get'
+import { InternalEntity } from 'src/glossary'
+import { OrderBy, SortDirection } from './queryTypes'
+
+const log = debug('sortResults')
+
+type FlatSortCriteria = [string[], SortDirection]
+
+export function warnOnIneffectiveSortingKeys(obj: Record<string, any>): void {
```

Comment:
I see a lot of 'any' in this codebase, but generally 'unknown' is safer
```suggestion
export function warnOnIneffectiveSortingKeys(obj: Record<string, unknown>): void {
```
There are legitimate uses for 'any' but as a rule of thumb you should start with 'unknown' until you are sure you need 'any'.

Reply:
That's a valid concern. I believe `any` was adopted because `unknown` throws type violations in a strict (non-strict?) mode for some users.